community/kernel/src/main/java/org/neo4j/kernel/recovery/LogTailScanner.java

```diff
@@ -155,6 +162,11 @@ else if ( entry instanceof LogEntryStart )
                 oldestStartEntryTransaction, oldestVersionFound, highestLogVersion, latestLogEntryVersion );
     }
 
+    private boolean hasUnreadableBytes( LogVersionedStoreChannel channel, long maxEntryReadEndPosition ) throws IOException
+    {
+        return channel.position() > maxEntryReadEndPosition;
```

Comment:
So this is based on the assumption that the `ReadAheadLogChannel` reads ahead in chunks and if stops before reading all the bytes that it read ahead from the underlying channel, i.e. some bad entry was encountered, right?

Will that always be safe to assume? What if the entry cursor reads something bad right at the end of a chunk, before it reads ahead to the next chunk? Wouldn't we fail to recognize that if only doing this comparison?

Reply:
we will check for the difference between position that channel had after last successful entry read and position after read ahead cursor was exhausted. That should show the difference only when we have partial record in the end of the file only and not between chunks inside read ahead cursor.